### PR TITLE
Slack updates

### PR DESCRIPTION
**Pull request by liorblob**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21419
https://github.com/demisto/etc/issues/22742

## Description
Adjust writing to integration context
fixed a bug where messages were not sent to the dedicated channel for notifications

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests

### CODE REVIEW

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 1]
```diff
@@ -653,13 +658,13 @@ def check_for_answers():
                 }
                 user = send_slack_request_sync(CLIENT, 'users.info', http_verb='GET', body=body).get('user', {})
                 users.append(user)
-                set_to_latest_integration_context('users', users)
 
             answer_question(actions[0].get('text', {}).get('text'), question, questions,
                             user.get('profile', {}).get('email'))
 
     questions = list(filter(lambda q: q.get('remove', False) is False, questions))
```
Comment (glicht) [inline Packs/Slack/Integrations/Slack/Slack.py]: Need to refetch the questions before doing filter.

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 2]
```diff
@@ -744,7 +748,7 @@ def check_for_mirrors():
                 else:
                     demisto.info('Could not mirror {}'.format(mirror['investigation_id']))
 
-                set_to_latest_integration_context('mirrors', mirrors)
+                set_to_latest_integration_context({'mirrors': mirrors})
```
Comment (glicht) [inline Packs/Slack/Integrations/Slack/Slack.py]: Also here we need to fetch the mirrors again and merge with the updated.

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 3]
```diff
@@ -1204,15 +1210,31 @@ def api_call(method: str, http_verb: str = 'POST', file: dict = None, params=Non
...
+    # Set
+    mocker.patch.object(demisto, 'getIntegrationContext', side_effect=get_integration_context)
+    mocker.patch.object(demisto, 'setIntegrationContext', side_effect=set_integration_context)
+
+    # Arrange
+    check_for_mirrors()
+
+    # Assert
+    assert demisto.setIntegrationContext.call_count == 0
+
```
Comment (Itay4) [inline Packs/Slack/Integrations/Slack/Slack_test.py]: assert one call to getintegrationcontext

**Comment by kirbles19** [comment]
@liorblob Perfect, no changes.

**Pull request merged**